microcosm_pubsub/tests/test_sentry.py

```diff
@@ -0,0 +1,83 @@
+"""
+Sentry integration tests.
+
+"""
+from hamcrest import assert_that, equal_to, is_
+from microcosm.api import create_object_graph
+from sentry_sdk.utils import BadDsn
+
+from microcosm_pubsub.sentry import before_send
+from microcosm_pubsub.tests.sentry_fixture import sample_event
+
+
+def test_sentry_config_bad_dsn():
+    def loader(metadata):
+        return dict(
+            sentry_logging=dict(
+                enabled=True,
+                dsn="invalid schema",
+            ),
+            build_info_convention=dict(
+                sha1="some-sha",
+            )
+        )
+
+    graph = create_object_graph("example", testing=False, loader=loader)
+    try:
+        graph.sentry_logging.enabled
+        assert False
+    except BadDsn:
+        pass
+
+
+def test_sentry_client_loaded_to_graph_testing():
+    def loader(metadata):
+        return dict(
+            sentry_logging=dict(
+                dsn="topic",
+                enabled=True,
+            )
+        )
+
+    graph = create_object_graph("example", testing=True, loader=loader)
+    assert_that(
+        graph.sentry_logging.dsn,
+        is_(equal_to("topic"))
+    )
+
+
+def test_sentry_client_default_to_None_no_dsn_set():
+    def loader(metadata):
+        return dict()
+
+    graph = create_object_graph("example", testing=True, loader=loader)
+
+    assert_that(
+        graph.sentry_logging.enabled,
+        is_(equal_to(False))
+    )
+
+
+def test_before_send_data_removal():
+    filtered_event = before_send(sample_event, None)
+    filterd_stacktrace = filtered_event["exception"]["values"][0]["stacktrace"]["frames"]
+
+    assert_that(
+        filterd_stacktrace[0]["vars"],
+        is_(equal_to(
+            {'cls': "<class 'microcosm_pubsub.result.MessageHandlingResult'>",
+             'handler': '<function context_logger.<locals>.wrapped at 0x109422ef0>',
+             'message': '<redacted>'}
```

Comment:
for a test of redaction, I'd probably go for a simplified dict rather the big blob snapshot you've got.

Reply:
I started with that but the big blob test actually helped in resolving an issue (if `before_send` fails sentry error doesn't send and doesn't even show a console log error). Will see if I can remove some more of the cruft to make it more readable.